### ISSUE TITLE
Update-DbaInstance Restart feature to avoid boot storm

### ISSUE THREAD

**Issue by lostsole** [sqlcollaborative/dbatools#9561]
### Summarize Functionality

Can you enhance the -restart switch feature to stagger the rebooting of the servers so that they do not run in parallel sow we can avoid boot storm issue with allocation of resources to the VM's.

### Is there a command that is similiar or close to what you are looking for?

Yes

### Technical Details

I had to run 

  foreach($computer in $VarComputerlist){
  $restartParams = @{
        ComputerName = $computer
        ErrorAction  = 'Stop'
        For          = 'WinRM'
        Wait         = $true
        Force        = $true
    }
    
    try {
            $null = Restart-Computer @restartParams
            $restarted = $true
        } catch {
            Stop-Function -Message "Failed to restart computer" -ErrorRecord $_
        }
}
at the end of the dbatools to had to cycle through the list our sql servers one at a time that got patched, Because we would get this error.
There is insufficient system memory in resource pool 'internal' to run this query.
and
Error: 17300, Severity: 16, State: 1. (Params:). The error is printed in terse mode because there was error during formatting. Tracing, ETW, notifications etc are skipped.

**Comment by niphlod** [comment]
ahem, from a process standpoint .... you apply patches to 200 servers, servers NEEDS reboot, and you'd like to stagger reboots, increasing the downtime for the whole farm ?
wouldn't an approach like patching 10 servers at a time, with immediate reboot, be more sound and less disruptive ?

**Comment by lostsole** [comment]
yes that would work , but not all hosting services are the same,  so the number for 200 servers reality is over 30 causes the boot storm issues on our system, The team that responsible for the underlying hardware is looking into the issues to see if it is a boot storm issues.  The DBA team only has a short window to patch all the servers, didn't realize this would have been an issues, but I guess I could use the -Throttle feature to 10.  Since the default is 50. Or remove the restart as part of the patch and manually restart at the end the script.

**Comment by niphlod** [comment]
So, I don't get it .... you've been given a tool (Update-DbaInstance) that you can ALREADY tune to your liking, via:
- passing a "reduced" list of instances to patch, up until your infrastructure limits
- using a -Throttle number that won't impact your infrastructure
- avoid restarting (which I'm reaaaally not a fan of, if the server NEEDS a restart, the window between the server asking for a restart and the actual restart should be the smallest possible window)

and, still, you want yet another parameter to delay restarts (which is nonsense, given you can already delay the patch installation (which, too, impacts resources)).

dbatools can't replace "the DBA team", nor talk with your infrastructure/hosting services team. 
It gives them **tools** to figure out a **solution**.

**Comment by lostsole** [comment]
So I just put this out there.
As I after switching to use the DBAtools to patch the SQL servers , We started to get an issues with the patched servers.  And we tracked this down to the Restarts, since they all restarted at the same time.
Since I don't know what the limit of our underlying hardware yet on what the max servers that can restart in parallel.

So I agree that I have solutions now that I have talked and you asked the questions and gave the answers.

I was only trying to get the -restart to have it's own Parallel -Throttle setting.  Just an idea.
